core/src/main/java/de/danoeh/antennapod/core/sync/SyncService.java

```diff
@@ -109,6 +114,47 @@ public Result doWork() {
             }
             return Result.retry();
         }
+
+
+    }
+
+    private ISyncService getActiveSyncProvider() {
+        String selectedService = getSelectedSyncProviderKey(getApplicationContext());
+        switch (selectedService) {
+            case SYNC_PROVIDER_CHOICE_GPODDER_NET:
+                return new GpodnetService(AntennapodHttpClient.getHttpClient(),
+                        GpodnetPreferences.getHosturl(), GpodnetPreferences.getDeviceID(),
+                        GpodnetPreferences.getUsername(), GpodnetPreferences.getPassword());
+            case SYNC_PROVIDER_CHOICE_NEXTCLOUD:
+                return new NextcloudSyncService(getApplicationContext(), ClientConfig.USER_AGENT);
+            default:
+                return null;
+        }
+    }
+
+
+    public static void setIsProviderConnected(boolean isConnected) {
+        getSyncServiceSharedPreferences()
+                .edit()
+                .putBoolean(SHARED_PREFERENCE_IS_SYNC_PROVIDER_CONNECTED, isConnected)
+                .apply();
+    }
+
+    public static boolean isProviderConnected() {
+        return getSyncServiceSharedPreferences()
+                .getBoolean(SHARED_PREFERENCE_IS_SYNC_PROVIDER_CONNECTED, false);
+    }
+
+    public static void setSelectedSyncProvider(String userSelect) {
```

Comment:
i now reordered methods so that public and private methods do not mix.

There is not method `getSelectedSyncProvider` but i assume you mean `getActiveSyncProvider()`. They are not close to each other as `setSelectedSyncProvider()` is a public method and `getActiveSyncProvider()` is not